### ISSUE TITLE
Fully support STJ's converters

### ISSUE THREAD

**Issue by angelaki** [domaindrivendev/Swashbuckle.AspNetCore#2050]
I guess that's a pretty interessting problem and I have no idea wether to ask here or at Microsoft's STJ's repo. But since it's more Swashbuckle-related I'll give it a try here.

Right now interface serialization is almost not supported by STJ, so I've written converters that fully handle the serialization itself. Sure Swashbuckle **can't** even know about it so it generated models like the default serializer would do. Newtonsoft had the ability to generate model descriptors based on a type and Swashbuckle took advantage if it. But since STJ once serializes "on demand", I really see no chance at all to respect custom Converters. How is / was it done with the EnumConverter - I guess the most common one?

If I had to take care of this issue ... I'd be completely lost ... Guess it's more a "wait for interface serialization, we can't take care of converters"?

**Comment by domaindrivendev** [comment]
Yep custom converters are tricky be because the serialization semantics are captured in runtime code as opposed to static constructs (serializer settings, type reflection etc.) that can be inspected by Swashbuckle. For enums, we can enumerate all possible values and simply run through the serializer but this isn't really possible for complex types/interfaces when the set of possible values is infinite.
